utils/frame/try-runtime/cli/src/lib.rs

```diff
@@ -328,10 +340,116 @@ where
 	Ok(())
 }
 
+async fn execute_block<Block, ExecDispatch>(
+	shared: SharedParams,
+	command: ExecuteBlockCmd,
+	config: Configuration,
+)-> sc_cli::Result<()>
+where
+	Block: BlockT + serde::de::DeserializeOwned,
+	Block::Hash: FromStr,
+	<Block::Hash as FromStr>::Err: Debug,
+	NumberFor<Block>: FromStr,
+	<NumberFor<Block> as FromStr>::Err: Debug,
+	ExecDispatch: NativeExecutionDispatch + 'static,
+{
+	let wasm_method = shared.wasm_method;
+	let execution = shared.execution;
+	let heap_pages = if shared.heap_pages.is_some() {
+		shared.heap_pages
+	} else {
+		config.default_heap_pages
+	};
+
+	let mut changes = Default::default();
+	let max_runtime_instances = config.max_runtime_instances;
+	let executor = NativeExecutor::<ExecDispatch>::new(
+		wasm_method.into(),
+		heap_pages,
+		max_runtime_instances,
+	);
+
+	let (mode, block) = match command.state {
+		State::Snap{ snapshot_path, url } => {
+			// `--header-at` is a required option for `snap`
+			let header_hash: Block::Hash = command.header_at
+				.expect("`--header-at` is a required option for `execute-block snap`.")
+				.parse()
+				.map_err(|e| format!("Could not parse header hash: {:?}", e))?;
+			let block: Block = rpc_api::get_block::<Block, _>(url.clone(), header_hash).await?;
+
+			let mode = Mode::Offline(OfflineConfig {
+				state_snapshot: SnapshotConfig::new(snapshot_path),
+			});
+
+			(mode, block)
+		},
+		State::Live { url, snapshot_path, block_at, modules } => {
+			let block_hash: Block::Hash = block_at
+				.ok_or("execute-block requires `block_at` option")?
```

Comment:
This is a better pattern than `expect`, but it's likely still worth having structopt enforce the requirement here.